pytalibstream/atr_keeper.py

```diff
@@ -0,0 +1,45 @@
+from typing import Optional
+import pyslidingwindow
+from pytalibstream.types.hlc import HLC
+
+
+class AtrKeeper:
+    def __init__(self, max_len: Optional[int] = 10):
+        if max_len < 2:
+            raise Exception(
+                "AtrKeeper period must be >= 2")
+
+        self.high = pyslidingwindow.SlidingWindowFloat(max_len)
+        self.low = pyslidingwindow.SlidingWindowFloat(max_len)
+        self.close = pyslidingwindow.SlidingWindowFloat(max_len)
+        self._atr = 0
+        self.data_count = 0
+        self.max_len = max_len
+        self.prev_tr = []
+
+    def add(self, candle: HLC):
+        self.data_count += 1
+        self.high.push(candle.high)
+        self.low.push(candle.low)
+        self.close.push(candle.close)
+
+        if self.max_len + 1 > self.data_count > 1:
+            self.prev_tr.append(self.tr)
```

Comment:
it looks like we should only append when `data_count > 1`
because if there is only one element, the true range will not be accurate.

true range depends on  at least two values.

Reply:
please compare the original node js code logic

Reply:
`if (this.dataLen < this.period + 1) {
      if (this.dataLen > 1) {
        this.previousTr.push(this.getTr());
      }
    }`

This is the original node js code. I might be overlooking something but the idea here was to avoid nested if statements for readability.

`self.data_count > 1`

This part of the condition should take care of appending only when there is an element